### PR TITLE
refactor + enhancement

### PR DESCRIPTION
**Pull request by malik672**
## Summary by CodeRabbit

- **New Features**
	- Updated the Uniswap SDK dependency to version 1.0.0-rc for enhanced features and performance.
	- Added a new `rustc-hash` dependency for improved hash map performance.
	- Enhanced `README.md` documentation with a "Getting Started" section and usage guidelines.
	- Introduced a new `Router` struct for managing trading pairs and finding optimal trade paths.

- **Bug Fixes**
	- Improved error handling and reporting across various entities, including `Pair`, `Trade`, and `Route`.

- **Tests**
	- Added comprehensive unit tests for the `Pair`, `Route`, and `Trade` entities to ensure functionality and correctness.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,4 +20,5 @@ anyhow = "1.0"
 once_cell = "1.19"
 ruint = "1.11"
 thiserror = "1.0"
-uniswap-sdk-core = "0.21.0"
+uniswap-sdk-core = "0.26.0"
```
Comment (shuhuiluo) [inline Cargo.toml]:
```suggestion
uniswap-sdk-core = "1.0.0-rc"
```

File: Cargo.toml [hunk 2]
```diff
@@ -20,4 +20,5 @@ anyhow = "1.0"
...
+rustc-hash = "2.0"
```
Comment (shuhuiluo) [inline Cargo.toml:24]: Add new line at eof.

**Review by shuhuiluo — CHANGES_REQUESTED** [review]
I don't know the point of this PR with 1000 lines of change in a single commit. There's not even a description if not thanks to CodeRabbit. Break this up to meaningful PRs.
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests

**Comment by malik672** [comment]
> I don't know the point of this PR with 1000 lines of change in a single commit. There's not even a description if not thanks to CodeRabbit. Break this up to meaningful PRs. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests

😭😭 just wanted to pull everything at once

**Pull request closed without merging**